crates/consensus/beacon/src/engine/sync.rs

```diff
@@ -304,7 +304,15 @@ where
 
         // drain an element of the block buffer if there are any
         if let Some(block) = self.range_buffered_blocks.pop() {
-            return Poll::Ready(EngineSyncEvent::FetchedFullBlock(block.0 .0))
+            // peek ahead and pop duplicates
+            while let Some(peek) = self.range_buffered_blocks.peek() {
+                if peek.0 .0.hash() == block.0 .0.hash() {
+                    self.range_buffered_blocks.pop();
+                } else {
+                    break;
+                }
+            }
```

Comment:
please use peek_mut

besides that, this is exactly what we want

Reply:
Sorry, not sure how to use that here - if I just use peek_mut and then try to do peek.pop() I'm getting an error that the method doesn't exist. 
Seems like `PeekMut` struct does have a pop method so I think I'm missing a feature being enabled or something like that. Please let me know if you know how to fix this.